src/gwt/src/org/rstudio/studio/client/application/ApplicationThemes.java

```diff
@@ -103,23 +133,30 @@ public void onComputeThemeColors(ComputeThemeColorsEvent event)
          Debug.logException(e);
       }
       
-      // Send the server the computed colors
-      server_.setComputedThemeColors(foreground, background, new VoidServerRequestCallback());
+      // update cached theme colors
+      themeColors_ = JsObject.createJsObject();
+      themeColors_.setString("foreground", foreground);
+      themeColors_.setString("background", background);
+      
+      // force eager refresh of client state after changing themes
+      events_.fireEvent(new PushClientStateEvent(true));
    }
 
    @Override
    public void onThemeChanged(ThemeChangedEvent event)
    {
-      RStudioThemes.initializeThemes(userPrefs_.get(),
-                                     userState_.get(),
-                                     Document.get(),
-                                     root_);
+      RStudioThemes.initializeThemes(userPrefs_.get(), userState_.get(), Document.get(), root_);
+      
+      // Wait a small amount of time for the theme to be applied
+      Timers.singleShot(1000, () -> {
```

Comment:
I'd really like a reliable way of forcing a synchronous render of CSS styles but I haven't found one that works 😞